crates/chain/src/tx_data_traits.rs

```diff
@@ -100,24 +100,42 @@ impl Anchor for ConfirmationBlockTime {
     }
 }
 
-/// An [`Anchor`] that can be constructed from a given block, block height and transaction position
-/// within the block.
-pub trait AnchorFromBlockPosition: Anchor {
-    /// Construct the anchor from a given `block`, block height and `tx_pos` within the block.
-    fn from_block_position(block: &bitcoin::Block, block_id: BlockId, tx_pos: usize) -> Self;
+/// Minimal set of parameters that are needed to construct a generic [`Anchor`].
+/// Typically used as an additional constraint on anchor:
+/// `A: Anchor + From<BlockPosition<'b>>`.
+pub struct BlockPosition<'b> {
+    block: &'b bitcoin::Block,
+    block_id: BlockId,
 }
 
-impl AnchorFromBlockPosition for BlockId {
-    fn from_block_position(_block: &bitcoin::Block, block_id: BlockId, _tx_pos: usize) -> Self {
-        block_id
+impl<'b> BlockPosition<'b> {
+    /// Creates new [`BlockPosition`].
+    pub fn new(block: &'b bitcoin::Block, block_id: BlockId) -> Self {
```

Comment:
Shouldn't the block hash just come from the block here. Pass the height in only.